### PR TITLE
Readthedocs conda environment

### LINKED ISSUE #1510: ReadTheDocs fails to build on Latest
Error message
```
Installed /home/docs/checkouts/readthedocs.org/user_builds/ga4gh-reference-implementation/envs/latest/lib/python2.7/site-packages/ga4gh-0.3.5.dev78+ng73e496b-py2.7.egg
Processing dependencies for ga4gh==0.3.5.dev78+ng73e496b
error: protobuf 3.1.0.post1 is installed but protobuf==3.0.0b3 is required by set(['ga4gh-client', 'ga4gh-schemas'])
```

### PR DESCRIPTION
**Pull request by david4096**
Adds the file needed to control the environment using conda (we weren't previously).

Uses the git tag in `docs/environment.yml` to control which dependencies to build against.

Also uses the PR https://github.com/ga4gh/schemas/pull/781 that checks compiled protobuf into the tree. Before we can be completely satisfied with this release we need to test installation in an environment without `protoc` installed.

You can see the RTD build [here](https://readthedocs.org/projects/server-1/builds/4915443/) and the result [here](http://server-1.readthedocs.io/en/latest/datarepo.html#command-line).

Close #1510

### CODE REVIEW

File: constraints.txt [hunk 1]
```diff
@@ -24,6 +24,6 @@
 # ga4gh-client==0.1.0
 # ga4gh-schemas==0.1.0
 #
-git+git://github.com/ga4gh/schemas.git@master#egg=ga4gh_schemas
+git+git://github.com/dcolligan/schemas.git@david_774#egg=ga4gh_schemas
```
Comment (dcolligan) [inline constraints.txt]: I presume this will be reverted after david_774 is merged into schemas/master?

File: docs/environment.yml [hunk 2]
```diff
@@ -13,5 +13,9 @@ dependencies:
 - wheel=0.29.0=py27_0
 - zlib=1.2.8=3
 - pip:
-    - git+git://github.com/ga4gh/ga4gh-client.git@master#egg=ga4gh_client --upgrade
-    - git+git://github.com/ga4gh/schemas.git@master#egg=ga4gh_schemas --upgrade
+    - git+git://github.com/ga4gh/ga4gh-client.git@master#egg=ga4gh_client
+    - git+git://github.com/david4096/schemas.git@fixpip#egg=ga4gh_schemas
```
Comment (dcolligan) [inline docs/environment.yml]: similar comment here

File: constraints.txt (hunk 1 above)
Comment (david4096) [inline constraints.txt]: Yeah, it seems natural that when a server PR is dependent on a schema change, the merger can update the dependencies as the merges come down.

**Comment by david4096** [comment]
@dcolligan Thanks, yes, I don't plan on ever having something other than `master` in the checked in constraints.

Using conda packaging is not a bad idea, but this PR is simply intended to address RTD. However, in principle the same environment can be used elsewhere. Offering the server in the conda packages would be a good route to go.

**Comment by dcolligan** [comment]
This is 8 commits... could you squash them down to 1?

**Comment by kozbo** [comment]
@dcolligan Github now has a "squash and merge" option in the pull down on the merge button. We have been using that lately.

**Pull request merged**